### PR TITLE
Support for refs and defs

### LINKED ISSUE #5: Add support for references
$defs and $ref aren't well supported within library. Library would ignore references that are empty (and ignore refs)

### PR DESCRIPTION
**Pull request by jackdelahunt**
Closes #5 

### Verification Steps

1. Do x
2. Do y

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change
- [ ] Other (please specify)

### CODE REVIEW

File: pkg/surveyjson/schema_type.go [hunk 1]
```diff
@@ -46,7 +46,7 @@ type JSONSchemaType struct {
 
 	Not *JSONSchemaType `json:"not,omitempty"`
 
-	Definitions Definitions `json:"definitions,omitempty"`
+	Definitions Definitions `json:"$defs,omitempty"`
```
Comment (wtrocki) [inline pkg/surveyjson/schema_type.go]: definitions is official name and it should stay. It is just that connectors team uses custom prefix

https://pkg.go.dev/github.com/swaggest/jsonschema-go#DefinitionsPrefix

Comment (jackdelahunt) [reply pkg/surveyjson/schema_type.go]: agh okay makes sense 👍

Comment (wtrocki) [reply pkg/surveyjson/schema_type.go]: If we want that to be done we cannot rely on the parser (as those expect specific key) and have separate function that user can supply that will be collecting definitions (by key etc.)

https://github.com/swaggest/jsonschema-go/blob/v0.3.35/context.go#L12

Comment (wtrocki) [reply pkg/surveyjson/schema_type.go]: Ok.. if that is another use case apart from "definitions" I think we can define them both in our schema parser:
```
Definitions "json:definitions"
DefinitionsAlias "json:$defs"
```
That would do the job

File: pkg/surveyjson/survey.go [hunk 2]
```diff
@@ -328,10 +345,37 @@ func (o *JSONSchemaOptions) recurse(name string, prefixes []string, requiredFiel
...
 }
 
+func parseRefPath(path string) ([]string, error) {
+
+	if strings.HasPrefix(path, RefPathPrefixDefs) {
+		path = strings.TrimPrefix(path, RefPathPrefixDefs)
+	} else if strings.HasPrefix(path, RefPathPrefixDefinitions) {
+		path = strings.TrimPrefix(path, RefPathPrefixDefinitions)
+	} else {
+		return nil, errors.New(fmt.Sprintf("Reference path must start with definition prefix: %v or %v", RefPathPrefixDefs, RefPathPrefixDefinitions))
```
Comment (wtrocki) [inline pkg/surveyjson/survey.go:373]: Love it!

**Review by wtrocki — APPROVED** [review]

**Comment by wtrocki** [comment]
We need to get some unit tests for those cases and also enums. That can happen in separate PR

**Comment by jackdelahunt** [comment]
looking into unit tests now. Never go unit tests in go so should be fun

**Comment by wtrocki** [comment]
@jackdelahunt That can be separate PR. Let's give it a go in the RHOAS CLI.

After enabling this feature I found separate issue #9

**Pull request merged**